network/f5/bigip_virtual_server.py

```diff
@@ -0,0 +1,438 @@
+    - "F5 developed module 'bigsuds' required (see http://devcentral.f5.com)"
+    - "Best run as a local_action in your playbook"
+requirements:
+    - bigsuds
+options:
+    server:
+        description:
+            - BIG-IP host
+        required: true
+    user:
+        description:
+            - BIG-IP username
+        required: true
+    password:
+        description:
+            - BIG-IP password
+        required: true
+    validate_certs:
+        description:
+            - If C(no), SSL certificates will not be validated. This should only be used
+              on personally controlled sites using self-signed certificates.
+        required: false
+        default: 'yes'
+        choices: ['yes', 'no']
+    state:
+        description:
+            - Pool member state
+        required: false
+        default: present
+        choices: ['present', 'absent', 'enabled', 'disabled']
+        aliases: []
+    partition:
+        description:
+            - Partition
+        required: false
+        default: 'Common'
+    name:
+        description:
+            - "Virtual server name."
+        required: true
+        aliases: ['vs']
+    destination:
+        description:
+            - "Destination IP of the virtual server (only host is currently supported) . Required when state=present and vs does not exist. Error when state=absent."
+        required: true
+        aliases: ['address', 'ip']
+    port:
+        description:
+            - "Port of the virtual server . Required when state=present and vs does not exist"
+        required: true
+    all_profiles:
+        description:
+            - "List of all Profiles (HTTP,ClientSSL,ServerSSL,etc) that must be used by the virtual server"
+        required: false
+        default: null
+    pool:
+        description:
+            - "Default pool for the virtual server"
+        required: false
+        default: null
+    snat:
+        description:
+            - "Source network address policy"
+        required: false
+        default: None
+    default_persistence_profile:
+        description:
+            - "Default Profile which manages the session persistence"
+        required: false
+        default: None
+    description:
+        description:
+            - "Virtual server description."
+        required: false
+        default: null
+'''
+
+EXAMPLES = '''
+
+## playbook task examples:
+
+---
+# file bigip-test.yml
+# ...
+  - name: Add VS
+    local_action:
+        module: bigip_virtual_server
+        server: lb.mydomain.net
+        user: admin
+        password: secret
+        state: present
+        partition: MyPartition
+        name: myvirtualserver
+        destination: "{{ ansible_default_ipv4["address"] }}"
+        port: 443
+        pool: "{{ mypool }}"
+        snat: Automap
+        description: Test Virtual Server
+        all_profiles:
+            - http
+            - clientssl
+
+  - name: Modify Port of the Virtual Server
```

Comment:
I changed the code so that when the VS is updated : 
+ If destination is present and port absent => destination is updated
+ if port is present and destination is absent => port is updated
+ if both are present => both are updated

It is more logic